activerecord/lib/active_record/relation/predicate_builder.rb

```diff
@@ -99,24 +99,26 @@ def expand_from_hash(attributes, &block)
           elsif value.is_a?(Hash) && !table.has_column?(key)
             table.associated_table(key, &block)
               .predicate_builder.expand_from_hash(value.stringify_keys)
-          elsif table.associated_with?(key)
+          elsif (associated_reflection = table.associated_with?(key))
```

Comment:
Using a boolean method to get the value is a breaking the expectation of what this method means. Let's use the non-`?` version, or rename it since it is nodoc.

Reply:
https://github.com/rails/rails/pull/55700